nav2_behaviors/include/nav2_behaviors/plugins/drive_on_heading.hpp

```diff
@@ -146,6 +175,8 @@ class DriveOnHeading : public TimedBehavior<ActionT>
       return ResultStatus{Status::FAILED, ActionT::Result::COLLISION_AHEAD};
     }
 
+    last_vel_ = std::make_shared<double>(cmd_vel->twist.linear.x);
```

Comment:
This is weird. If you want to initialize it so you know its valid, you could default it to `numerical_limits<double>::max()` and reset to that / check for it if we have a last value. I don't like this smart pointer stuff though